### ISSUE TITLE
Intermittent failure creating map with persistence

### ISSUE THREAD

**Issue by mickyloo** [OpenHFT/Chronicle-Map#354]
We are getting intermittent failures during map creation in our CI tests.  
The test creates a small <LongValue, LongValue> map of 5 entries with persistence to file in a temp folder.

```
java.io.IOException: Not enough space to fallocate 0 MiB to / unallocated was 0 MiB
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.fallocate(VanillaChronicleHash.java:1078)
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.map(VanillaChronicleHash.java:1058)
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.createMappedStoreAndSegments(VanillaChronicleHash.java:511)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithNewFile(ChronicleMapBuilder.java:1850)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithFile(ChronicleMapBuilder.java:1748)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createPersistedTo(ChronicleMapBuilder.java:1593)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createOrRecoverPersistedTo(ChronicleMapBuilder.java:1614)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createOrRecoverPersistedTo(ChronicleMapBuilder.java:1604)
	at service.data.chronicle.ChronicleBuilder.build(ChronicleBuilder.java:75)
	at service.data.chronicle.ChronicleBuilderTest.build_ToFile_whenValueTypeIsLong(ChronicleBuilderTest.java:65)
```

chronicle-map version 3.21.86
JDK8 
Debian

We are not seeing this issue in 3.20.84.1

**Comment by alamar** [comment]
Is it possible that you are running out of disk space @mickyloo ?

**Comment by mickyloo** [comment]
@alamar We have confirmed that our pod does have available disk space when we see that error.

**Comment by minborg** [comment]
Are you able to share the actual test that is failing?

**Comment by minborg** [comment]
I have run my test over 50k times and was unable to reproduce:

![image](https://user-images.githubusercontent.com/7457876/141496352-a766306f-2fac-4032-9a88-0e036875b564.png)

**Comment by mickyloo** [comment]
This is the minimal code that allows us to reproduce the issue intermittently.
```
public class ChronicleTest {
  @Rule
  public TemporaryFolder testFolder = new TemporaryFolder();

  @Test
  public void build_toFile() throws IOException {
    String baseDirectory = testFolder.getRoot().toString();
    File file = new File(baseDirectory, "chronicle.dat");
    ChronicleMap<LongValue, LongValue> map = ChronicleMapBuilder.of(LongValue.class, LongValue.class)
        .name("test")
        .entries(5)
        .createOrRecoverPersistedTo(file, true);

    assertTrue(file.isFile());
  }
}
```

**Comment by minborg** [comment]
Hi @mickyloo 

A while ago, we reworked the handling of the recover operations and made them more rigid. One of the things we did was to acquire an exclusive lock during recovery. It should be mentioned that the recover operations should only be invoked "manually" when making sure there are no other processes using the map.

Is it possible for you to run a more recent version (e.g. BOM 2.22ea57)?

I have re-run you test above over 50 000 times with no problems using the BOM above. 

 
![image](https://user-images.githubusercontent.com/7457876/141826924-ce0c5b6d-d325-4567-bd7f-3dafab403660.png)

**Comment by minborg** [comment]
Did you update your version @mickyloo ?

**Comment by mickyloo** [comment]
We still see the issue with the latest version, 3.22ea5.

**Comment by minborg** [comment]
I have added tests that try to reproduce your problem. When they pass on all our build platforms, this issue will be automatically closed. Please reopen if the problem persists @mickyloo

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.22ea6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.22ea6), [BOM-2.22ea86](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea86)
